src/Flow/ETL/Transformer/EntryValueTransformer.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Transformer;
+
+use Flow\ETL\Exception\RuntimeException;
+use Flow\ETL\Row;
+use Flow\ETL\Row\EntryFactory;
+use Flow\ETL\Row\Factory\NativeEntryFactory;
+use Flow\ETL\Rows;
+use Flow\ETL\Transformer;
+
+/**
+ * @psalm-immutable
+ */
+final class EntryValueTransformer implements Transformer
+{
+    private static ?bool $isSerializable = null;
+
+    /**
+     * @var array<string>
+     */
+    private array $entries;
+
+    private string $userFunction;
+
+    private EntryFactory $entryFactory;
+
+    /**
+     * @param array<string> $entries
+     * @param string $userFunction
+     */
+    public function __construct(array $entries, string $userFunction, EntryFactory $entryFactory = null)
+    {
+        $this->entries = $entries;
+        $this->userFunction = $userFunction;
+        $this->entryFactory = $entryFactory ?? new NativeEntryFactory();
+    }
+
+    /**
+     * @return array{entries: array<string>, user_function: string, entry_factory: EntryFactory}
+     */
+    public function __serialize() : array
+    {
+        /** @psalm-suppress ImpureMethodCall */
+        if (!self::isSerializable()) {
+            throw new RuntimeException('CallbackEntryTransformer is not serializable without "opis/closure" library in your dependencies.');
+        }
+
+        return [
+            'entries' => $this->entries,
+            'user_function' => $this->userFunction,
+            'entry_factory' => $this->entryFactory,
+        ];
+    }
+
+    /**
+     * @param array{entries: array<string>, user_function: string, entry_factory: EntryFactory} $data
+     * @psalm-suppress MoreSpecificImplementedParamType
+     */
+    public function __unserialize(array $data) : void
+    {
+        /** @psalm-suppress ImpureMethodCall */
+        if (!self::isSerializable()) {
```

Comment:
this is not correct, if callable is a regular string or an array, serialization is possible even without opis/closure